src/common/snippets/src/pass/gn_tokenization.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/gn_tokenization.hpp"
+#include "snippets/pass/collapse_subgraph.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "snippets/utils.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+ov::snippets::pass::TokenizeGNSnippets::TokenizeGNSnippets() {
+    MATCHER_SCOPE(TokenizeGNSnippets);
+
+    auto group_norm_pattern = ov::pass::pattern::wrap_type<ov::op::v12::GroupNormalization>();
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::pass::TokenizeGNSnippets")
+        auto group_norm_node = ov::as_type_ptr<ov::op::v12::GroupNormalization>(m.get_match_root());
+        if (group_norm_node->is_dynamic() || group_norm_node->get_element_type() != element::f32)
```

Comment:
I have some investigation on other precisions previously with this convert solution.
1. the data flow and LIR is expected for other precisions.
2. have run tests with bf16 input with snippets. It passed with a relatively larger abs_threshold(such as 0.01).
3. have run tests with f32 input with bf16 infer_precision with snippets. It passed with a relatively larger abs_threshol.
4. try to run i8 input, seems like reference gn implementation does not support int8, so not run successfully.

Reply:
I am asking because we have to distigush between codegen limitations and performance limitations caused by backend. So ideaally such conditions on fp32 precision should be applied in backed specific callback of GNTokenization transform

Reply:
To this, we need apply you suggestion "[move GNDecomposition pass after Snippets tokenization](https://github.com/openvinotoolkit/openvino/pull/21751#discussion_r1544195323)" below first. See my [reply](https://github.com/openvinotoolkit/openvino/pull/21751#discussion_r1545935552) please for the explanation. Thank you Dmitry for the good idea!